sdks/python/apache_beam/yaml/yaml_transform_test.py

```diff
@@ -68,6 +68,25 @@ def test_chain_with_input(self):
           ''')
       assert_that(result, equal_to([41, 43, 47, 53, 61, 71, 83, 97, 113, 131]))
 
+  def test_chain_with_source_sink(self):
+    with beam.Pipeline(options=beam.options.pipeline_options.PipelineOptions(
+        pickle_library='cloudpickle')) as p:
+      elements = p | beam.Create(range(10))
+      result = elements | YamlTransform(
+          '''
+          type: chain
+          source:
+            type: Create
+            elements: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+          transforms:
+            - type: PyMap
+              fn: "lambda x: x * x + x"
+          sink:
```

Comment:
Sinks can have outputs (e.g. the actual files written or other metadata). Here I'm using a map as the sink to be able to test it.